packages/core/src/shared/errors.ts

```diff
@@ -584,3 +588,17 @@ export function isNetworkError(err?: unknown): err is Error & { code: string } {
         'EADDRNOTAVAIL', // port not available/allowed?
     ].includes(err.code)
 }
+
+/**
+ * This error occurs on a network call if the user has set up a proxy in the
+ * VS Code settings but the proxy is not reachable.
+ *
+ * Setting ID: http.proxy
+ */
+function isVSCodeProxyError(err?: unknown): boolean {
```

Comment:
Not sure why `unknown` is needed in some of these functions, but at least for most of these we can force the caller to make that decision.
```suggestion
function isVSCodeProxyError(err?: Error): boolean {
```

Reply:
the `err` comes from a `catch` and is unknown, so this just makes it easier to throw it in and not worry about it